lsp-mode.el

```diff
@@ -3474,7 +3482,12 @@ in that particular folder."
 
     (when (and lsp-enable-links
                (lsp-feature? "textDocument/documentLink"))
-      (add-hook 'lsp-on-idle-hook #'lsp--document-links nil t)))
+      (add-hook 'lsp-on-idle-hook #'lsp--document-links nil t))
+
+    (when (and lsp-enable-dap-auto-configure
+               (fboundp 'dap-mode)
+               (not (bound-and-true-p dap-mode)))
```

Comment:
I think we can change this to use `featurep`. With current logic, even if user disable `dap-mode`, it will reenable itself again next time a buffer is opened. And user has no choice but set `lsp-enable-dap-auto-configure` to `nil`. Which can be tedious.
```suggestion
               (featurep 'dap-mode)
```